RoaringBitmap/src/main/java/org/roaringbitmap/BitSetUtil.java

```diff
@@ -71,6 +72,71 @@ public static RoaringBitmap bitmapOf(final long[] words) {
     return ans;
   }
 
+  // To avoid memory allocation, reuse ThreadLocal buffers
+  private static final ThreadLocal<long[]> WORD_BLOCK = ThreadLocal.withInitial(() ->
+      new long[BLOCK_LENGTH]);
+
+  /**
+   * Efficiently generate a RoaringBitmap from an uncompressed byte array or ByteBuffer
+   * This method tries to minimise all kinds of memory allocation
+   *
+   * @param bb the uncompressed bitmap
+   * @param fastRank if set, returned bitmap is of type
+   *                 {@link org.roaringbitmap.FastRankRoaringBitmap}
+   * @return roaring bitmap
+   */
+  public static RoaringBitmap bitmapOf(ByteBuffer bb, boolean fastRank) {
+
+    bb = bb.slice().order(ByteOrder.LITTLE_ENDIAN);
+    final RoaringBitmap ans = fastRank ? new FastRankRoaringBitmap() : new RoaringBitmap();
+
+    // split buffer into blocks of long[], reuse a ThreadLocal array for blocks
+    final long[] words = WORD_BLOCK.get();
+    int containerIndex = 0;
+    int blockLength = 0, blockCardinality = 0, offset = 0;
+    long word;
+    while (bb.remaining() >= 8) {
+      word = bb.getLong();
+
+      // Add read long to block
+      words[blockLength++] = word;
+      blockCardinality += Long.bitCount(word);
+
+      // When block is full, add block to bitmap
+      if (blockLength == BLOCK_LENGTH) {
+        // Each block becomes a single container, if any bit is set
+        if (blockCardinality > 0) {
+          ans.highLowContainer.insertNewKeyValueAt(containerIndex++, Util.highbits(offset),
+              BitSetUtil.containerOf(0, blockLength, blockCardinality, words));
+        }
+        offset += (BLOCK_LENGTH * Long.SIZE);
```

Comment:
Though this can be reasonably dismissed, there is the possibility that offset overflows. Make sure that the offset variable cannot overflow (hopefully it cannot due to the the max size of a Java Bitset, but please be specific, maybe with a comment).